examples/DGmethods/held_suarez_balanced.jl

```diff
@@ -0,0 +1,567 @@
+    end
+    S[_dρe] = 0
+  end
+end
+#md nothing # hide
+
+# This defines the local wave speed from the current state (this will be needed
+# to define the numerical flux)
+function wavespeed(n, Q, aux, _...)
+  @inbounds begin
+    ρ_ref, ρe_ref, ϕ = aux[_a_ρ_ref], aux[_a_ρe_ref], aux[_a_ϕ]
+    dρ, ρu, ρv, ρw, dρe = Q[_dρ], Q[_ρu], Q[_ρv], Q[_ρw], Q[_dρe]
+
+    ## get total energy and density
+    ρ = ρ_ref + dρ
+    e = (ρe_ref + dρe) / ρ
+
+    ## velocity field
+    u, v, w = ρu / ρ, ρv / ρ, ρw / ρ
+
+    ## internal energy
+    e_int = e - (u^2 + v^2 + w^2)/2 - ϕ
+
+    ## compute the temperature
+    T = air_temperature(e_int)
+
+    abs(n[1] * u + n[2] * v + n[3] * w) + soundspeed_air(T)
+  end
+end
+#md nothing # hide
+
+# The only boundary condition needed for this test problem is the no flux
+# boundary condition, the state for which is defined below. This function
+# defines the plus-side (exterior) values from the minus-side (inside) values.
+# This plus-side value will then be fed into the numerical flux routine in order
+# to enforce the boundary condition.
+function nofluxbc!(QP, _, _, nM, QM, _, auxM, _...)
+  @inbounds begin
+    DFloat = eltype(QM)
+    ## get the minus values
+    dρM, ρuM, ρvM, ρwM, dρeM = QM[_dρ], QM[_ρu], QM[_ρv], QM[_ρw], QM[_dρe]
+
+    ## scalars are preserved
+    dρP, dρeP = dρM, dρeM
+
+    ## vectors are reflected
+    nx, ny, nz = nM[1], nM[2], nM[3]
+
+    ## reflect velocities
+    mag_ρu⃗ = nx * ρuM + ny * ρvM + nz * ρwM
+    ρuP = ρuM - 2mag_ρu⃗ * nx
+    ρvP = ρvM - 2mag_ρu⃗ * ny
+    ρwP = ρwM - 2mag_ρu⃗ * nz
+
+    ## Construct QP state
+    QP[_dρ], QP[_ρu], QP[_ρv], QP[_ρw], QP[_dρe] = dρP, ρuP, ρvP, ρwP, dρeP
+  end
+end
+#md nothing # hide
+
+#------------------------------------------------------------------------------
+
+# ### Definition of the problem
+# Here we define the initial condition as well as the auxiliary state (which
+# contains the reference state on which the initial condition depends)
+
+# First it is useful to have a conversion function going between Cartesian and
+# spherical coordinates (defined here in terms of radians)
+function cartesian_to_spherical(DFloat, x, y, z)
+    r = hypot(x, y, z)
+    λ = atan(y, x)
+    φ = asin(z / r)
+    (r, λ, φ)
+end
+#md nothing # hide
+
+# FXG: reference state
+# Setup the reference state based on a N=0.0158725 uniformly stratified atmosphere with θ0=315K
+function auxiliary_state_initialization!(T0, aux, x, y, z)
+  @inbounds begin
+    DFloat = eltype(aux)
+    p0 = DFloat(MSLP)
+    θ0 = DFloat(315)
+    N_bv=DFloat(0.0158725)
+    gravity=DFloat(grav)
+    
+    ## Convert to Spherical coordinates
+    (r, λ, φ) = cartesian_to_spherical(DFloat, x, y, z)
+
+    ## Calculate the geopotential ϕ
+    h = r - DFloat(planet_radius) # height above the planet surface
+    ϕ = gravity * h
+
+    ## Reference Potential Temperature from constant Brunt-Vaisala definition
+    θ_ref = θ0 * exp( N_bv^2*h/gravity )
```

Comment:
Was this `θ0` on purpose? If so, why not `θ_ref = DFloat(0)`?

Reply:
Not sure I understand, `θ0` is defined as`θ0 = DFloat(315)` above .

Reply:
Oops, haha, I got confused by the syntax highlighting and forgot that variable names are not commutative (`θ0 != 0θ`)!